src/components/history-and-comments.component.tsx

```diff
@@ -1,81 +1,89 @@
 import React from "react";
-import { DataTableSkeleton, Button, Form, TextInput } from "@carbon/react";
-import { OverflowMenuVertical } from "@carbon/react/icons";
+import {
+  DataTableSkeleton,
+  OverflowMenu,
+  OverflowMenuItem,
+} from "@carbon/react";
 import { useTranslation } from "react-i18next";
 import { parseDate, formatDatetime } from "@openmrs/esm-framework";
 import styles from "./history-and-comments.scss";
 import { usePrescriptionDetails } from "../medication-request/medication-request.resource";
+import { deleteMedicationDispense } from "../medication-dispense/medication-dispense.resource";
 import MedicationEventCard from "./medication-event-card.component";
 
-const HistoryAndComments: React.FC<{ encounterUuid: string }> = ({
-  encounterUuid,
-}) => {
+const HistoryAndComments: React.FC<{
+  encounterUuid: string;
+  mutatePrescriptionTableRows: Function;
+}> = ({ encounterUuid, mutatePrescriptionTableRows }) => {
   const { t } = useTranslation();
-  const { requests, dispenses, prescriptionDate, isError, isLoading } =
+  const { requests, dispenses, mutate, prescriptionDate, isError, isLoading } =
     usePrescriptionDetails(encounterUuid);
 
-  // TODO: assumption is dispenses always are after requqests?
+  const generateMedicationDispenseActionMenu: Function = (
+    medicationDispenseUuid: string
+  ) => (
+    <OverflowMenu
+      ariaLabel="Medication Dispense Action Menu"
+      flipped={true}
+      className={styles.medicationEventActionMenu}
+    >
+      <OverflowMenuItem
+        onClick={() => {
+          deleteMedicationDispense(medicationDispenseUuid);
+          mutate();
+          mutatePrescriptionTableRows();
+        }}
+        itemText={t("delete", "Delete")}
+      ></OverflowMenuItem>
+    </OverflowMenu>
+  );
+
+  // TODO: assumption is dispenses always are after requests?
   return (
     <div className={styles.historyAndCommentsContainer}>
       {isLoading && <DataTableSkeleton role="progressbar" />}
       {isError && <p>Error</p>}
       {dispenses &&
-        dispenses.map((medication) => {
+        dispenses.map((dispense) => {
           return (
-            <div>
+            <div key={dispense.id}>
```

Comment:
I went through and added a bunch of keys to suppress those key warnings in the console logs....

Reply:
👍🏾